include/zephyr/bluetooth/conn.h

```diff
@@ -262,6 +262,139 @@ enum __packed bt_conn_type {
 			   BT_CONN_TYPE_SCO | BT_CONN_TYPE_ISO,
 };
 
+/** Remote channel sounding capabilities for LE connections supporting CS */
+struct bt_conn_le_cs_capabilities {
+	/** Number of CS configurations */
+	uint8_t num_config_supported;
+	/** Maximum number of consecutive CS procedures.
+	 *
+	 * When set to zero, indicates support for both fixed and indefinite
+	 * numbers of CS procedures before termination.
+	 */
+	uint16_t max_consecutive_procedures_supported;
+	/** Number of antennas. */
+	uint8_t num_antennas_supported;
+	/** Maximum number of antenna paths. */
+	uint8_t max_antenna_paths_supported;
+	/** Initiator role. */
+	bool initiator_supported;
+	/** Reflector role. */
+	bool reflector_supported;
+	/** Mode-3 */
+	bool mode_3_supported;
+	/** RTT AA-Only */
+	bool rtt_aa_only_supported;
+	/** RTT Sounding */
+	bool rtt_sounding_supported;
+	/** RTT Random Payload */
+	bool rtt_random_payload_supported;
+	/** RTT AA Only accuracy.
+	 *
+	 * - True: 10ns time-of-flight precision
+	 * - False: 150ns time-of-flight precision (or not supported)
+	 */
+	bool rtt_aa_only_10ns_accurate;
+	/** RTT Sounding accuracy.
+	 *
+	 * - True: 10ns time-of-flight precision
+	 * - False: 150ns time-of-flight precision (or not supported)
+	 */
+	bool rtt_sounding_10ns_accurate;
+	/** RTT Random Payload accuracy
+	 *
+	 * - True: 10ns time-of-flight precision
+	 * - False: 150ns time-of-flight precision (or not supported)
+	 */
+	bool rtt_random_payload_10ns_accurate;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT AA Only.
+	 *
+	 * Set to 0 if RTT AA Only isn't supported.
+	 */
+	uint8_t rtt_aa_only_n;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT Sounding.
+	 *
+	 * Set to 0 if RTT Sounding isn't supported
+	 */
+	uint8_t rtt_sounding_n;
+	/** Number of CS steps needed to achieve the
+	 * accuracy requirements for RTT Random Payload.
+	 *
+	 * Set to 0 if RTT Random Payload isn't supported.
+	 */
+	uint8_t rtt_random_payload_n;
+	/** Phase-based normalized attack detector metric
+	 * when a CS_SYNC with sounding sequence is received.
+	 */
+	bool phase_based_nadm_sounding_supported;
+	/** Phase-based normalized attack detector metric
+	 * when a CS_SYNC with random sequence is received.
+	 */
+	bool phase_based_nadm_random_supported;
+	/** CS_SYNC LE 2M PHY. */
+	bool cs_sync_2m_phy_supported;
+	/** CS_SYNC LE 2M 2BT PHY. */
+	bool cs_sync_2m_2bt_phy_supported;
+	/** CS Subfeatures.
+	 *
+	 * - Bit 0: CS with no Frequency Actuation Error.
+	 * - Bit 1: CS Channel Selection Algorithm #3c
+	 * - Bit 2: CS phase-based ranging from RTT sounding sequence.
+	 */
+	uint16_t subfeatures_supported;
```

Comment:
It is a bit inconsistent that we stopped mapping the bits into booleans beyond this member. Is it intentional?

Should we instead always use the full value and just add bitmask macros?

Reply:
That's a fair point. What's the rationale here?